library/psa_crypto.c

```diff
@@ -3499,32 +3499,38 @@ uint32_t mbedtls_psa_interruptible_get_max_ops( void )
 }
 
 uint32_t mbedtls_psa_sign_hash_get_num_ops(
-            const mbedtls_psa_sign_hash_interruptible_operation_t *operation )
+                    mbedtls_psa_sign_hash_interruptible_operation_t *operation)
 {
 #if (defined(MBEDTLS_PSA_BUILTIN_ALG_ECDSA) || \
-     defined(MBEDTLS_PSA_BUILTIN_ALG_DETERMINISTIC_ECDSA) ) && \
-     defined( MBEDTLS_ECP_RESTARTABLE )
+    defined(MBEDTLS_PSA_BUILTIN_ALG_DETERMINISTIC_ECDSA)) && \
+    defined(MBEDTLS_ECP_RESTARTABLE)
 
-    return( operation->restart_ctx.ecp.ops_done );
+    /* Hide the fact that the restart context only holds a delta of number of
+     * ops done during the last operation, not an absolute value. */
+    operation->num_ops += operation->restart_ctx.ecp.ops_done;
```

Comment:
This won't work if the application calls `complete()` twice before `get_num_ops()`, or calls `get_num_ops()` twice in a row. You need to update the ops count after each call to the underlying function.

Please fix (blocker) and add non-regression tests. I'm ok with deferring the tests if `operation` goes back to being `const`.

Reply:
#7127